clang/lib/3C/DeclRewriter.cpp

```diff
@@ -723,8 +734,7 @@ void FunctionDeclBuilder::buildDeclVar(const FVComponentVariable *CV,
     if (Range.isValid() && !inParamMultiDecl(PVD) ) {
       Type = getSourceText(Range, *Context);
       if (!Type.empty()) {
-      // Great, we got the original source including any itype and bounds.
-        IType = "";
+        IType = getExistingIType(CV->getExternal()) + BoundsStr;
```

Comment:
Are we sure that it doesn't already have a bounds if it gets here? Maybe `BoundsStr` will have the existing one if it was there?

Reply:
It think comment I deleted was wrong, The `Type` string didn't contain the bounds. This just wasn't manifesting as a problem since it would only matter when rewriting a function where one parameter is an a array with bounds, and another parameter gets rewriting from unchecked to checked. e.g.,

```c
void foo(_Array_ptr<int> a  : count(1), int *b) { }
```

`BoundsStr` will contain the original bounds if there were bounds present in the source.